internal/driver/driver.go

```diff
@@ -166,6 +166,14 @@ func (c *Conn) WriteRaw(b []byte) error {
 
 // Request sends the given request to the connection and returns the response.
 func (c *Conn) Request(ctx context.Context, header *wire.MsgHeader, body wire.MsgBody) (*wire.MsgHeader, wire.MsgBody, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/4146
-	panic("not implemented")
+	if err := c.Write(header, body); err != nil {
+		return nil, nil, lazyerrors.Error(err)
+	}
+
+	header, body, err := c.Read()
+	if err != nil {
+		return nil, nil, lazyerrors.Error(err)
+	}
+
```

Comment:
At the very least, we should check that the message we read is actually a response (by checking RequestID / ResponseTo). Maybe do something else too